src/http/server.cr

```diff
@@ -138,8 +138,8 @@ class HTTP::Server
     end
   end
 
-  def bind
-    @server ||= TCPServer.new(@host, @port)
+  def bind(reuse_port = false)
```

Comment:
But we have like 8 constructors, it will clutter the API.

For me, having it as a property or as named arguments in `bind`/`listen` works well too. But I'd probably add it to both `listen` and `bind`, because it's not very common to do `bind` + `listen`, just `listen`.